app.js

```diff
@@ -11,7 +11,82 @@ app.use(express.json());
 app.use(bodyParser.json());
 
 
+app.post('/register', (req, res) => {
+  const userId = randomUUID();
+  const newUser = {
+    id: userId,
+    email: "alex.gmail.com",
+    pass: "[pxnjyfgbcfnm"
+  }
+  res.set({'x-user-id': userId})
+  users.push(newUser);
+  res.status(201).send(users);
 
+});
+
+app.get('/products', (req, res) => {
+  res.send(products);
+});
+
+app.get('/products/:id', (req, res) => {
+  const product = products.find(p => p.id == req.params.id )
+  if (!product) return res.status(404).send("Product not found!")
+  res.send(product)
+});
+
+app.put('/cart/:id', (req, res) => {
+
+  const userId = req.header('x-user-id');
+
+  if (!userId || !users.find(u => u.id === userId)) {
+    return res.status(401).send({ error: 'Unauthorized' });
+  }
+
+  if (!carts[userId]) {
+  carts[userId] = [];
```

Comment:
`carts ` is an Array not an Object, userId is UUID(example: 550e8400-e29b-41d4-a716-446655440000), so it can not be used as array index